src/typegate/engine/bindings.ts

```diff
@@ -454,3 +456,43 @@ export function archive(a0: ArchiveInp): ArchiveResult {
     return { Err: { message: err.toString() } };
   }
 }
+
+export type CreateOrGetResult =
+  | {
+      Ok: {
+        run: Run;
+      };
+    }
+  | {
+      Err: {
+        message: string;
+      };
+    };
+
+export type PersistRunResult =
+  | {
+      Ok: string;
+    }
+  | {
+      Err: {
+        message: string;
+      };
+    };
+
+export function createOrGetRun(inp: CreateOrGetInput): CreateOrGetResult {
```

Comment:
thought: these binding functions aren't exactly necessary. Feel free to use the methods on the Meta object directly. Should work as long as the types in runtime.d.ts are good.

Reply:
Resolved in #845.